server/src/main/java/io/crate/types/BitStringType.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.types;
+
+import java.io.IOException;
+import java.util.Base64;
+import java.util.BitSet;
+import java.util.List;
+import java.util.Locale;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import io.crate.Streamer;
+import io.crate.sql.tree.BitString;
+
+public final class BitStringType extends DataType<BitString> implements Streamer<BitString>, FixedWidthType {
+
+    public static final int ID = 25;
+    // TODO: better max length limit, or use a `ZERO` sentinel?
+    // SortedDocValuesField max is 32766 bytes
```

Comment:
This restriction could be lifted by allowing to disable the docvalues (columnstore=off) for BitString types (same which we already support for `text`). Of course we would have to document the trade-off.

Reply:
I was more concerned about the values leading to long segment merges and thought that it would be better to add a (artificial) lower limit to prevent these from happening.

Even if we'd allow to disable the doc-values there would still be the issue of the inverse index, which could potentially blow up with large a large number of very large unique terms. 

Not sure what's the best way to handle this.

Reply:
The `32766 bytes` limitation would only be lifted if the docvalues and indexing is turned off.
But the point about long segment merges is still valid,  I'm also fine to leave that out for now and come back to it if really needed.